### PR TITLE
Add unit tests for Administration Rules section

### PR DESCRIPTION
**Pull request by jeff-phillips-18**
Refactored code for testability:
  -  Moved bulk of the view control code into new rulesListDirective
  -  Changed RuleState methods to return the promise from the CollectionsAPI
     rather than creating a new promise.

Coverage Before:
====================== Coverage summary ======================
Statements   : 30.83% ( 1493/4842 )
Branches     : 8.22% ( 128/1557 )
Functions    : 25.71% ( 326/1268 )
Lines        : 30.83% ( 1493/4842 )
===========================================================

Coverage After:
====================== Coverage summary ======================
Statements   : 36.19% ( 1708/4719 )
Branches     : 10.88% ( 163/1498 )
Functions    : 30.91% ( 387/1252 )
Lines        : 36.19% ( 1708/4719 )
===========================================================

### CODE REVIEW

File: tests/rules/save-rule-modal-service.spec.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
+/* jshint -W117, -W030 */
```
Comment (AllenBW) [inline tests/rules/save-rule-modal-service.spec.js]: no need for jshint comments

File: tests/rules/rules-state.spec.js [hunk 2]
```diff
@@ -0,0 +1,127 @@
+/* jshint -W117, -W030 */
```
Comment (AllenBW) [inline tests/rules/rules-state.spec.js]: no need for jshint comments

File: client/app/states/administration/rules/rules-list.directive.js [hunk 3]
```diff
@@ -0,0 +1,368 @@
+/* eslint camelcase: "off" */
+(function() {
+  'use strict';
+
+  angular.module('app.states')
```
Comment (AllenBW) [inline client/app/states/administration/rules/rules-list.directive.js]: given this is a directive it probably should be in the components folder and attached to `app.components`

File: client/app/states/administration/rules/rules-list.directive.js [hunk 4]
```diff
@@ -0,0 +1,368 @@
...
+        },
+        controller: RulesListController,
+        controllerAs: 'vm',
+        bindToController: true,
+      };
+    });
+
+  /** @ngInject */
+  function RulesListController(RulesState, SaveRuleModal, $state, $rootScope, $timeout, $log, $scope) {
+    /* jshint validthis: true */
```
Comment (AllenBW) [inline client/app/states/administration/rules/rules-list.directive.js]: unnecessary jshint comment

File: tests/rules/rules-list.directive.spec.js [hunk 5]
```diff
@@ -0,0 +1,277 @@
+/* jshint -W117, -W030 */
```
Comment (AllenBW) [inline tests/rules/rules-list.directive.spec.js]: jshint comment not necessary

File: tests/rules/rules-list.directive.spec.js [hunk 6]
```diff
@@ -0,0 +1,277 @@
...
+describe('app.states.RulesListDirective', function() {
```
Comment (AllenBW) [inline tests/rules/rules-list.directive.spec.js]: if ya move it, gonna have to update this `app.components`

File: tests/rules/rules-state.service.spec.js [hunk 7]
```diff
@@ -0,0 +1,130 @@
+/* jshint -W117, -W030 */
```
Comment (AllenBW) [inline tests/rules/rules-state.service.spec.js]: unnecessary jshint comment

**Comment by jeff-phillips-18** [comment]
@AllenBW Thanks for the comments. Addressed your concerns.

**Comment by chriskacerguis** [comment]
Looks like still a few CodeClimate issues that need to be addressed.

**Comment by jeff-phillips-18** [comment]
@chriskacerguis Any pointer on how to figure out what it is that CodeClimate is eluding to?

**Comment by chriskacerguis** [comment]
@jeff-phillips-18 click on "details", 

https://codeclimate.com/github/ManageIQ/manageiq-ui-self_service/pull/191

**Comment by jeff-phillips-18** [comment]
@chriskacerguis Sorry, I followed the details link, on that page I've clicked every link I could think might help but they all seem to lead to the same page 😟

**Comment by chriskacerguis** [comment]
@jeff-phillips-18 do you see this page?
screen shot 2016-09-26 at 4 36 47 pm

**Comment by jeff-phillips-18** [comment]
@chriskacerguis Yes. I don't understand the issues and I'm not finding the links to be helpful.

**Comment by chriskacerguis** [comment]
@jeff-phillips-18 on each issue there is a read more info.  Take a look there, it seems that there is a lot of duplicate code, that can probably be streamlined.  Take a look and let me know if you still need help and @AllenBW or I can help

**Comment by jeff-phillips-18** [comment]
@chriskacerguis Still not that useful but I *think I know what it's getting at. This will likely make the code less readable in places. I've installed Docker and codeclimate so that I can run and check for issues prior to pushing. The default of 15 minutes to run the analysis is too short and I've updated it to 30 minutes...

**Comment by jeff-phillips-18** [comment]
I believe all code climate issues have been fixed now

File: tests/rules/rules-list.directive.spec.js [hunk 8]
```diff
@@ -0,0 +1,276 @@
...
+  var compileHTML = function (markup, scope) {
+    element = angular.element(markup);
+    $compile(element)(scope);
+
+    scope.$digest();
+  };
+
+  beforeEach(function () {
+    Session.create({
+      auth_token: 'b10ee568ac7b5d4efbc09a6b62cb99b8',
```
Comment (chriskacerguis) [inline tests/rules/rules-list.directive.spec.js]: I have not dug into this, so take this at face value, but do we want a token hard coded (again, not sure if this is doesn't matter here)

Comment (jeff-phillips-18) [inline tests/rules/rules-list.directive.spec.js]: This is in the test itself to get past any login requirements. I simply copied/pasted from another test spec.

Comment (chriskacerguis) [inline tests/rules/rules-list.directive.spec.js]: cools.  Just wanted to make sure 😄

**Pull request merged**